### PR TITLE
Drop Python 3.6 testing for ST2 3.9.0

### PR DESCRIPTION
**Pull request by philipphomberger**
Update Pipeline for Circleci.

### CODE REVIEW

**Comment by nzlosh** [comment]
@philipphomberger Is this MR ready for merge?

**Comment by philipphomberger** [comment]
> @philipphomberger Is this MR ready for merge?

Yes I think so. 

Would be very helpful to update the PIP dependencies in small units.

**Review by nzlosh — APPROVED** [review]

**Comment by nzlosh** [comment]
Good stuff!  Yes, we need to remove py3.6 testing to so we can start moving forward with adding mongodb and newer python versions.

**Review by amanda11 — COMMENTED** [review]
I am just wondering on best order to go about this...
We still have CI pipelines and build projects that are going to build python 3.6 images.
So it might be better to merge this PR after we have stopped the st2-packages and st2packaging-dockerfiles from requiring python 3.6 images.

**Comment by nzlosh** [comment]
There's already work underway to remove py3.6 from st2 packaging which should arrive this week or the next.

**Comment by cognifloyd** [comment]
I just merged #6118 which included dropping python3.6 from GHA. Sorry I didn't see your PR, or I would have merged this first.

**Review by cognifloyd — APPROVED** [review]

**Review by cognifloyd — CHANGES_REQUESTED** [review]
Would you move the changelog entry up to the in development section and adjust the wording to reflect the narrowed scope of this PR? Then I'll merge.

**Comment by philipphomberger** [comment]
> Would you move the changelog entry up to the in development section and adjust the wording to reflect the narrowed scope of this PR? Then I'll merge.

done

File: CHANGELOG.rst [hunk 1]
```diff
@@ -12,6 +12,8 @@ Fixed
 Changed
 ~~~~~~~
 * Bumped `jsonschema` 2.6.0 -> 3.2.0 now that python3.6 is not supported. #6118
+* Drop Python 3.6. Support. Update Pipeline Templates. #6080
```
Comment (cognifloyd) [inline CHANGELOG.rst:15]:
```suggestion
* Drop Python 3.6 testing in CircleCI. #6080
```

**Review by cognifloyd — APPROVED** [review]

**Pull request merged**